spring-integration-file/src/main/java/org/springframework/integration/file/splitter/FileSplitter.java

```diff
@@ -199,11 +206,23 @@ public Object next() {
 					return new FileMarker(filePath, Mark.END);
 				}
 				try {
-					return bufferedReader.readLine();
+					if (this.line == null && !this.done) {
+						this.line = bufferedReader.readLine();
+					}
```

Comment:
Sorry, need more time to figure out the proper algorithm, but I won't mind if you push something.
Thanks!